plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/metadata/MetadataRebuilder.java

```diff
@@ -238,23 +233,11 @@ public void end() {
      * Process exits from group level, executed in isolation.
      */
     private void rebuildMetadataExitGroup(final String currentGroupId) {
-      storageFacet.perform(db, new Operation<Void>()
```

Comment:
The scope of the retry is the outermost surrounding `@Transactional`, so in `metadataUpdater.processMetadata(...)` it will retry when updating the metadata (still on the fence whether we want to retry for non-update case) and will repeat the read+write, ie. the code inside the method marked with `@Transactional`.

For the `rebuildMetadataInner` case since there is another `@Transactional` method surrounding the code between `onEnterBaseVersion` and `onExitBaseVersion` that piece of code will be retried as a whole rather than separately retrying the code in `metadataUpdater.processMetadata` when exiting the base version.